### ISSUE TITLE
[Bug]: Open overlay drawer fails with Vite app

### ISSUE THREAD

**Issue by Narvalex** [microsoft/fluentui#31631]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (12) x64 11th Gen Intel(R) Core(TM) i5-11260H @ 2.60GHz
    Memory: 11.10 GB / 23.73 GB
  Browsers:
    Edge: Chromium (125.0.2535.92)
    Internet Explorer: 11.0.22621.3527
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://github.com/Narvalex/fluent-ui-9-drawer-overlay-bug/

## Actual Behavior
When opening the drawer an error throws in the console log.

## Expected Behavior
To open the drawer overlay without getting errors in dev mode

### Logs

```shell
console.js:273 
 @fluentui/react-utilities [useControllableState]:
A component must be either controlled or uncontrolled (specify either the state or the defaultState, but not both).
Decide between using a controlled or uncontrolled component and remove one of this props.
More info: https://reactjs.org/link/controlled-components
Error
    at useControllableState (http://localhost:5173/node_modules/.vite/deps/@fluentui_react-components.js?v=f3f30d38:5363:3)
    at useDialog_unstable (http://localhost:5173/node_modules/.vite/deps/@fluentui_react-components.js?v=f3f30d38:58055:27)
    at http://localhost:5173/node_modules/.vite/deps/@fluentui_react-components.js?v=f3f30d38:59474:17
    at renderWithHooks (http://localhost:5173/node_modules/.vite/deps/chunk-ZEDUCUT7.js?v=f3f30d38:11566:26)
    at updateFunctionComponent (http://localhost:5173/node_modules/.vite/deps/chunk-ZEDUCUT7.js?v=f3f30d38:14600:28)
    at updateSimpleMemoComponent (http://localhost:5173/node_modules/.vite/deps/chunk-ZEDUCUT7.js?v=f3f30d38:14481:18)
    at updateMemoComponent (http://localhost:5173/node_modules/.vite/deps/chunk-ZEDUCUT7.js?v=f3f30d38:14384:22)
    at beginWork (http://localhost:5173/node_modules/.vite/deps/chunk-ZEDUCUT7.js?v=f3f30d38:15995:22)
    at beginWork$1 (http://localhost:5173/node_modules/.vite/deps/chunk-ZEDUCUT7.js?v=f3f30d38:19779:22)
    at performUnitOfWork (http://localhost:5173/node_modules/.vite/deps/chunk-ZEDUCUT7.js?v=f3f30d38:19224:20)
overrideMethod	@	console.js:273
useControllableState	@	useControllableState.js:25
useDialog_unstable	@	useDialog.js:17
(anónimo)	@	Dialog.js:12
renderWithHooks	@	react-dom.development.js:15486
updateFunctionComponent	@	react-dom.development.js:19617
updateSimpleMemoComponent	@	react-dom.development.js:19454
updateMemoComponent	@	react-dom.development.js:19303
beginWork	@	react-dom.development.js:21712
beginWork$1	@	react-dom.development.js:27465
performUnitOfWork	@	react-dom.development.js:26596
workLoopSync	@	react-dom.development.js:26505
renderRootSync	@	react-dom.development.js:26473
performSyncWorkOnRoot	@	react-dom.development.js:26124
flushSyncCallbacks	@	react-dom.development.js:12042
(anónimo)	@	react-dom.development.js:25690
```

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by Narvalex** [comment]
I tried to replicate with create-react-app, but no error was thrown. In previous versions of Fluent UI  I did not have this issue

**Comment by miroslavstastny** [comment]
This has been fixed in #31435. Does not log the error since `@fluentui/react-components@9.53.0`.
